lib/liquid/block_body.rb

```diff
@@ -44,7 +43,7 @@ def parse(tokenizer, parse_context)
           end
           parse_context.trim_whitespace = false
           @nodelist << token
-          @blank &&= !!(token =~ WhitespaceOrNothing)
+          @blank &&= token.strip.empty?
```

Comment:
`strip` allocates a new string, so I wouldn't expect it to help much.  Instead, we could leverage String#match? if available, which doesn't allocate any objects.  We could even use refinements to make that more natural to use (e.g. `/\A\s*\z/.match?(token)`).

Reply:
While `String#match?` is fast, its only available with Ruby 2.4 and above,
I'm not sure how using refinements would be better, so feel free to overwrite this branch or replace with another PR..